src/yaml_input.c

```diff
@@ -873,12 +878,36 @@ static PetscErrorCode ValidateConfig(MPI_Comm comm, RDyConfig *config, PetscBool
   }
 
   // validate output options
-  PetscCheck((config->output.format == OUTPUT_NONE) || (config->output.interval > 0), comm, PETSC_ERR_USER,
-             "Output interval must be specified as a positive number of steps.");
-  PetscCheck((config->output.batch_size == 0) || (config->output.format != OUTPUT_BINARY), comm, PETSC_ERR_USER,
-             "Binary output does not support output batching");
-  if ((config->output.batch_size == 0) && (config->output.format != OUTPUT_NONE) && config->output.format != OUTPUT_BINARY) {
-    config->output.batch_size = 1;
+  if (config->output.format != OUTPUT_NONE || config->output.step_interval > 0 || config->output.time_interval > 0.0) {
+    config->output.enable           = PETSC_TRUE;
+    config->output.prev_output_time = -1.0;
+  } else {
+    config->output.enable = PETSC_FALSE;
+  }
+
+  if (config->output.enable) {
+    PetscCheck((config->output.format != OUTPUT_NONE), comm, PETSC_ERR_USER, "Output requested, but the format is not specified.");
+    PetscCheck(!(config->output.step_interval == 0 && config->output.time_interval == 0.0), comm, PETSC_ERR_USER,
+               "Output requested, but neither step_interval nor time_interval specified.");
+    PetscCheck((config->output.step_interval >= 0), comm, PETSC_ERR_USER, "Output step interval must be specified as a positive number of steps.");
+    PetscCheck((config->output.time_interval >= 0.0), comm, PETSC_ERR_USER, "Output time interval must be specified as a positive number of steps.");
+    PetscCheck((config->output.batch_size == 0) || (config->output.format != OUTPUT_BINARY), comm, PETSC_ERR_USER,
+               "Binary output does not support output batching");
+    if ((config->output.batch_size == 0) && (config->output.format != OUTPUT_NONE) && config->output.format != OUTPUT_BINARY) {
+      config->output.batch_size = 1;
+    }
+
+    if (config->output.time_interval) {
+      printf("config->time.coupling_interval = %f; coupling_interval_was_specified = %d\n", config->time.coupling_interval,
```

Comment:
Leftover print statement here. If we don't need this, you can probably get rid of `coupling_interval_was_specified`.